Source/ChartIQView.swift

```diff
@@ -916,7 +918,15 @@ extension ChartIQView: WKScriptMessageHandler {
     dataSource?.pullPaginationData(by: params, completionHandler: { [weak self] data in
       guard let self = self else { return }
       DispatchQueue.main.async {
-        self.formatJSQuoteData(data, cb: cb)
+        // Check to see if you need to try and retrieve more historical data.
+        // This is where you can put your own logic on when to stop retrieving historical data.
+        // By default if the last pagination request return 0 data then it has probably reached the end.
+        // If you have spotty data then another idea might be to check the last historical date, this would require you knowing what date to stop at though.
+        var moreAvailable = true
+        if(data.count < 1000) {
+            moreAvailable = false
+        }
+        self.formatJSQuoteData(data, moreAvailable: moreAvailable, cb: cb)
```

Comment:
I don't quite understand this. The data count is usually less than 1000 so at least in my case, moreAvailable is always false. But even when sending moreAvailable: false it still continues to query for additional pagination data when scrolling. Is that the

Reply:
I meant to check in 1 instead of 1000. I'll check the flag, it was working for me last week.